lib/url.c

```diff
@@ -2519,6 +2519,20 @@ static CURLcode parse_proxy(struct Curl_easy *data,
   }
 
   /* now, clone the proxy host name */
+#ifdef USE_UNIX_SOCKETS
+  if(proxytype == CURLPROXY_SOCKS5_HOSTNAME
+      && strncmp("unix/", proxy + 10, 5) == 0) { /* len("socks5h://") = 10 */
```

Comment:
The host name *might* start with that scheme, the type could also be set with a separate option and then the proxy name starts with the host name. This cannot assume the first 10 bytes are known.

I also find it a little unfortunate to reserve the `unix` host name like this, as there might actually be a proxy in use that is named like that. Maybe we should instead make the `--unix-socket` option accept a `socks...` prefix?

Reply:
> The host name might start with that scheme, the type could also be set with a separate option and then the proxy name starts with the host name. This cannot assume the first 10 bytes are known.

I will update to handle that case.

> there might actually be a proxy in use that is named like that.

In that case, `unix/path` cannot be a valid URI for a socks proxy as it can only have `host:port` syntax. Since I am checking for `unix/`, I think it should not affect anyone having a server named `unix` with a socks proxy running in it.

> Maybe we should instead make the --unix-socket option accept a socks... prefix

I wish to have the ability to set the proxy via environment variable like `ALL_PROXY=socks5h://unix/run/tor/socks`, so that programs that internally use curl does not have to change the command line.